app/src/main/java/com/blink22/android/mvpandroid/newTodo/NewTodoFragment.java

```diff
@@ -46,7 +50,10 @@ public static NewTodoFragment newInstance() {
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setPresenter(NewTodoPresenter.getInstance(new TodosSubscriber(NetworkManager.getInstance().getTodosService()), this));
+
+        ((BaseApp) getActivity().getApplication()).getNetworkComponent().inject(this);
+
+        setPresenter(NewTodoPresenter.getInstance(mTodosSubscriber, this));
```

Comment:
@AFGhazy this should be handled by DI, add `@Inject` to the presenter do the necessary changes (add @ provide .. etc) and remove this line.